cmake/checks/check_01_cxx_features.cmake

```diff
@@ -511,33 +511,32 @@ ENDIF()
 # We only run this check if we have CXX14 support, otherwise the use of constexpr
 # is limited (non-const constexpr functions for example).
 #
-IF(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-  CHECK_CXX_COMPILER_BUG(
-    "
-    #define Assert(x,y) if (!(x)) throw y;
-    void bar()
-    {}
+CHECK_CXX_COMPILER_BUG(
+  "
+  #define Assert(x,y) if (!(x)) throw y;
+  void bar()
+  {}
 
-    constexpr int
-    foo(const int n)
-    {
-      Assert(n>0, \"hello\");
-      if(!(n >= 0))
-        bar();
-      return n;
-    }
+  constexpr int
+  foo(const int n)
+  {
+    Assert(n>0, \"hello\");
+    if(!(n >= 0))
+      bar();
+    return n;
+  }
 
-    int main()
-    {
-      constexpr unsigned int n=foo(1);
-      return n;
-    }
-    "
-    DEAL_II_CXX14_CONSTEXPR_BUG)
-ENDIF()
+  int main()
+  {
+    constexpr unsigned int n=foo(1);
+    return n;
+  }
+  "
+  DEAL_II_CXX14_CONSTEXPR_BUG)
 
 SET(DEAL_II_CONSTEXPR "constexpr")
-IF(DEAL_II_CXX14_CONSTEXPR_BUG)
+# MSVC has considerable problems with "constexpr" as well
+IF(DEAL_II_CXX14_CONSTEXPR_BUG OR CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
```

Comment:
What https://en.cppreference.com/w/cpp/compiler_support#C.2B.2B14_library_features is lying? :wink:

Reply:
why was there no problem before?

Reply:
Yup :smile:

Reply:
I would just disable `DEAL_II_CXX14_CONSTEXPR_BUG` unconditionally for `MSVC`.